features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2

```diff
@@ -829,16 +829,34 @@
             <!--
                 FAPI Profile Support.
             -->
-            {% if oauth.oidc.fapi is defined %}
             <FAPI>
                 {% if oauth.oidc.fapi.enable_ciba_profile is defined %}
                 <EnableCibaProfile>{{oauth.oidc.fapi.enable_ciba_profile}}</EnableCibaProfile>
                 {% endif %}
                 {% if oauth.oidc.fapi.enable_security_profile is defined %}
                 <EnableSecurityProfile>{{oauth.oidc.fapi.enable_security_profile}}</EnableSecurityProfile>
                 {% endif %}
+                <SupportedClientAuthenticationMethods>
+                    {% if oauth.oidc.fapi.client_authentication_methods|length %}
+                        {% for authmethod in oauth.oidc.fapi.client_authentication_methods %}
+                            <SupportedClientAuthenticationMethod>{{authmethod}}</SupportedClientAuthenticationMethod>
+                        {% endfor %}
+                    {% else %}
+                        <SupportedClientAuthenticationMethod>private_key_jwt</SupportedClientAuthenticationMethod>
+                        <SupportedClientAuthenticationMethod>tls_client_auth</SupportedClientAuthenticationMethod>
+                    {% endif %}
+                </SupportedClientAuthenticationMethods>
+                <SupportedSignatureAlgorithms>
```

Comment:
In what places are these signature algorithms used?

Reply:
This is used at the token endpoint when validating the signature algorithm. If the algorithm is not specified in the app level, this is the fallback option. 
https://github.com/wso2-extensions/identity-oauth-addons/pull/108/files#diff-c1fefc87f2cf0485b2f5e352367c515cb1036459a5232698108ff48d4d50a5e1R760

@SachiniSiriwardene this will be used for DCR validations as well right?

Reply:
shall we make this config name specific to the place where we are using this (ex: token_endpoint_auth_signing_alg,  request_object_signing_alg , etc). Right now when we say supported it has broad usage.

Reply:
We made the name common and nested it under the <FAPI> parent tag because the spec mentions the same set of algorithms for both token endpoint and request object signing [1]. Therefore shall we keep the name as it is or do you feel that we need to change the config names based on the usage? WDYT?

[1] https://openid.net/specs/openid-financial-api-part-2-1_0.html#algorithm-considerations

Reply:
even though the spec mentions a common set of algorithms when an app registers them it is registered under different attributes right? And the usage is in specific unrelated cases, therefore I think it will be cleaner to maintain these separately.

Reply:
but still for validating we just need to check whether the algorithm is one of the above. It doesn't matter which use case it is (token assertion/request object). Those attributes will be stored separately. These values are just used to filter whether the sent attributes are one of the allowed ones for a FAPI application.